### ISSUE TITLE
eslint does not report when an "IIFE" is not invoked.

### ISSUE THREAD

**Issue by coreyfarrell** [eslint/eslint#8385]
**Tell us about your environment**

* **ESLint Version: v3.19.0**
* **Node Version: v6.10.0**
* **npm Version: v3.10.10**

**What parser (default, Babel-ESLint, etc.) are you using?**
I don't know so I assume default?

**Please show your full configuration:**
``` js
{
    "plugins": ["html"],
    "globals": {
        "Polymer": true
    },
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "no-console": "off",
        "no-extra-parens": ["error", "all"],
        "indent": ["error", "tab"],
        "linebreak-style": ["error", "unix"],
        "quotes": ["error", "single"],
        "semi": ["error", "always"]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
(function() {
	'use strict';
	/* private initialization code here */
});
```

**What did you expect to happen?**
I expected this code to raise an error:
* The anonymous function is unused as it isn't invoked or saved to a variable.
* Since the script above doesn't actually contain an IIFE this should have caused an error from the 'no-extra-parans' rule.

**What actually happened? Please include the actual, raw output from ESLint.**
Nothing, the script above produced no errors.

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, I think this is intended behavior given the rules that you have enabled. Although the anonymous function is unused, it's not a variable, so the [`no-unused-vars`](http://eslint.org/docs/rules/no-unused-vars) rule wouldn't be expected to report it. Even though the function isn't invoked, the parentheses around it are still necessary because removing them would create invalid syntax.

I would recommend enabling the [`no-unused-expressions`](http://eslint.org/docs/rules/no-unused-expressions) rule if you want to lint for code like this.

**Comment by coreyfarrell** [comment]
Thank you for the quick response.  Maybe the documentation should mention `"extends": "eslint:all"` as a way to test "very strict" and find issues not picked up by `eslint:recommended` rules?  I've done this and I'm now checking the 1000 errors produced, disabling many options, but also tweaking my code in response to other errors.  Either way I think this bug can be closed, I'm only leaving it open to make sure my comment about documentation is seen.

**Comment by not-an-aardvark** [comment]
The documentation does mention it [here](http://eslint.org/docs/user-guide/configuring#using-eslintall). We don't recommend using `eslint:all` in production, because it doesn't have any versioning guarantees. If we add a new rule in a minor release and you're using `eslint:all`, there's a good chance it will break your build. However, it is useful for exploring which rules are available (which seems to be what you're using it for).

**Comment by coreyfarrell** [comment]
Got it, thanks again.
